backend/backend/container.py

```diff
@@ -15,128 +24,175 @@ class Status(Enum):
+        """Removes storage volumes for all inactive (destroyed) containers
 
-    def destroy_container(self, id: int):
-        """Destroy a specified container
+        :param container_id: ID (name) of which container's storage directory to remove
+        """
+        self._client.volumes.prune()
+        logging.info("Pruned all unused volumes")
+
+    def get_volume(self, volume_id: str):
+        """Returns the specified volume in form <Volume: short_id>,
+        where short_id is the volume id truncated to 10 characters
 
-        :param id: ID (name) of container
+        :param volume_id: The name of the volume
         """
         try:
-            container_name = "openssh-server"+str(id)
-            self._client.containers.get(container_name).remove()
-        except:
-            raise Exception(
-                "Could not find or destroy the specified container")
+            return self._client.volumes.get(volume_id)
+        except Exception as exception:
+            raise exception
 
-    def status_container(self, id: int) -> Status:
-        """Return the status of a specific container with the ID argument
+    def status_container(self, container_id: str) -> Status:
+        """Return the status of a specific container with the container_id argument
 
-        :param id: ID (name) of container
+        :param container_id: ID (name) of container
         :return: Returns an enum describing the status of a container
         """
         try:
-            container_name = "openssh-server"+str(id)
-            sts = self._client.containers.get(
-                container_name).attrs['State']['Status']
-        except:
+            sts = self._client.containers.get(container_id).attrs['State']['Status']
+        except Exception:
             return Status.NOTFOUND
         else:
             if sts == "running":
                 return Status.RUNNING
-            elif sts == "exited":
-                return Status.EXITED
             elif sts == "restarting":
                 return Status.RESTARTING
             elif sts == "paused":
                 return Status.PAUSED
-            else:
-                return Status.UNDEFINED
-
-    def create_shared_folder(self, id: int, user: str):
-        """Creates a directory for the docker container with the specified id,
-        exposes the specified SSH port, and has SSH login credentials user/password.
-        Inside it creates a script for initialization that uses the specified username
-        for the home directory.
-
-        :param id: ID (name) of container
-        :param port: SSH port that is exposed
-        :param user: Username for SSH
-        :param password: Password for SSH
+            elif sts == "exited":
+                return Status.EXITED
+        return Status.UNDEFINED
+
+    def format_config(self, container_id: int, user: str, password: str,
+                      hostname='Dell-T140', user_id='1000', group_id='1000',
```

Comment:
Something I did not think of before was that these user:password combos that the frontend will be sending to here will obviously be insecure. By using them in our own container we are exposing our own containers to get bruteforced by attackers.

However I assume that the os port will be something random which would require attackers to scan the internet on this port to find the SSH server seems very unlikely so I think this is fine for now. In the final product we might want to use pub/priv keys instead of these unsecure credentials and have the custom init.sh script create the user account and set up the password. Maybe a new userstory?

Reply:
I gave that a thought as well, but I think I came to the same conclusion as you. The ports are random and the containers should only be online for a short amount of time, i expect. I believe that the risk of an attacker scanning the correct ip at the correct random port with the correct credentials is small.
That said, the documentation for our chosen image has support for keys, so it could be implemented if needed:  https://github.com/orgs/linuxserver/packages/container/package/openssh-server
Another option would be adding traffic rules to the Azure instance running the backend where we could add the addresses of our frontends to be the only allowed connections.